src/inflators/link.ts

```diff
@@ -2,12 +2,23 @@ import { addComponent } from "bitecs";
 import { Link } from "../bit-components";
 import { HubsWorld } from "../app";
 
+export enum LinkType {
+  LINK = 0,
+  AVATAR = 1,
+  SCENE = 2,
+  ROOM = 3,
+  ROOM_URL = 4,
+  WAYPOINT = 5
+}
+
 export type LinkParams = {
   href: string;
+  type?: LinkType;
 };
 
 export function inflateLink(world: HubsWorld, eid: number, params: LinkParams): number {
   addComponent(world, Link, eid);
   Link.url[eid] = APP.getSid(params.href);
+  Link.type[eid] = params.type !== undefined ? params.type : LinkType.LINK;
```

Comment:
If I understand correctly, `Link.type` can be asynchronously updated in link system depending on what url(src) actually points to.

`Link.type` can be accessed from link-hover-menu system (or future new systems) even before updating.

Isn't it confusing and less maintainable? Link label and click action can differ between before and after update the type.

Some random ideas to simplify.

1. We don't update type depending on what url(src) actually points to. We just respect Link component initial parameter. (Then we may need to update Hubs Blender-addon to force to specify type parameter properly?)
2. Introduce Link Initialize component and system and the system asynchronously resolves the url and then set Link component.

Any thoughts?

Reply:
If we want to avoid race and go with this PR, we may choose option 1 for now. And we may revisit later about auto detection in another PR. Even without auto type detection, supporting link types should be much more valuable than nothing.

Reply:
Yes that's true. This is based on the assumption that the link type is set by the links system very early when the interactable is spawned and between that and hovering there is going to be enough time to guess the type. This is clearly wrong because for cached assets the interactable can be spawned quite fast and in slow connections the type guessing can take more type and by that time the interactable can be already hovered.

Ragarding this:
> We don't update type depending on what url(src) actually points to. We just respect Link component initial parameter. (Then we may need to update Hubs Blender-addon to force to specify type parameter properly?)

I don't think we can do that as links are also used in spawned interactables (when you paste a link) and there there is no upfront type, we need to guess.

So I guess n.1 is the best option although we kind of are in the same race condition situation where we can't really show a link if the user hovers while the link type is still being initialized although that shouldn't take much time.

I think I'm going to add a comment there with all this info and address in a later PR as I don't have so much time now to work on this.

Reply:
> I don't think we can do that as links are also used in spawned HTML interactables (when you paste a link) and there there is no upfront type, we need to guess.

Then we may move the auto link type detection from `Link` to HTML pasting handling system. If we do that, `Link` system will be simpler. No async initialization will be needed and `Link` component data will be immutable.

Anyways, we may revisit it later. The current behavior in this PR is ok for now.